br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -984,8 +1093,9 @@ func (local *DupeController) CollectRemoteDuplicateRows(ctx context.Context, tbl
 	if err != nil {
 		return false, errors.Trace(err)
 	}
-	if err := duplicateManager.CollectDuplicateRowsFromTiKV(ctx, local.importClientFactory); err != nil {
-		return false, errors.Trace(err)
+	err = duplicateManager.CollectDuplicateRowsFromTiKV(ctx, local.importClientFactory, algorithm)
+	if err != nil {
+		return common.ErrFoundDataConflictRecords.Equal(err) || common.ErrFoundIndexConflictRecords.Equal(err), errors.Trace(err)
```

Comment:
see below comment

Reply:
I don' think `hasDup` is useless. If error happened, it may be other error like decoding failed or other failure, not necessarily means we encounter duplicate. We still need `common.ErrFoundDataConflictRecords.Equal(err) || common.ErrFoundIndexConflictRecords.Equal(err)` to determine whether we have duplicates.

`CollectDuplicateRowsFromTiKV` will not return `ErrFoundDuplicateKeys`. Here it can only return `ErrFoundDataConflictRecords` or `ErrFoundIndexConflictRecords`.